### DISCUSSION TITLE
How to change the MAUI DatePicker calendar view from English to french(other languages)

### DISCUSSION THREAD

**Discussion by saiganesh-sakthivel** [dotnet/maui#15753] [Q&A]
How to change the MAUI DatePicker calendar view from English to French (other languages). Even, i had try to set the culture info in the sample but it is not working. Any idea to achieve it?

**Comment by AndreasReitberger** [comment]
I set the `Culture` in the `App.xaml.cs` file like following.

```cs
public App()
{
        InitializeComponent();
       
       //....

        // Set language (culture your CultureInfo object`
        Thread.CurrentThread.CurrentCulture = culture;
        Thread.CurrentThread.CurrentUICulture = culture;
        Strings.Culture = culture;

}
```

This should also update all `Controls`

**Reply by saiganesh-sakthivel** [reply]
Hi @AndreasReitberger ,

<img width="539" alt="image" src="https://github.com/dotnet/maui/assets/61176526/d9454385-0bfe-4d6a-9366-6b4938b72a25">

name context 'Strings' is not available in maui, any namespace need to added it ah?

Regards,
SaiGanesh Sakthivel

**Reply by AndreasReitberger** [reply]
Hi @saiganesh-sakthivel 
`Strings` is my localized `Strings.resx` file. If you don't have such a file, you can comment this line ;)

**Reply by saiganesh-sakthivel** [reply]
Hi @AndreasReitberger ,

Thank you for the suggestions. Still facing the same issue even set the culture. MAUI DatePicker is not localized.

<img width="377" alt="image" src="https://github.com/dotnet/maui/assets/61176526/c56dc4b1-eb2b-4232-9a4f-47dd23b28047">

code snippet

public App()
	{
		InitializeComponent();
        MainPage = new AppShell();
        var culture = new CultureInfo("fr-FR"); // Replace with your desired culture or language code
        Thread.CurrentThread.CurrentCulture = culture;
        Thread.CurrentThread.CurrentUICulture = culture;
	}
	
	MainPage
	
	    <VerticalStackLayout
            Spacing="25"
            Padding="30,0"
            VerticalOptions="Center">
            <DatePicker />
        </VerticalStackLayout>

**Reply by vchelaru** [reply]
I am also having the same problem with date pickers not reflecting the current culture. I've tried setting:

```cs
var ci = new CultureInfo(cultureCode);

Thread.CurrentThread.CurrentCulture = ci;
Thread.CurrentThread.CurrentUICulture = ci;

CultureInfo.CurrentCulture = ci;
CultureInfo.CurrentUICulture = ci;

CultureInfo.DefaultThreadCurrentCulture = ci;
CultureInfo.DefaultThreadCurrentUICulture = ci;
```
